services/UserAccountService.ts

```diff
@@ -192,4 +193,72 @@ export default class UserAccountService {
       return userProfile;
     });
   }
+
+  /**
+   * This method is used to update the access levels of multiple users at once.
+   * @param accessUpdates - an array of objects containing the user and their new access level
+   * @param emails - an array of emails to update
+   * @param currentUser - the user making the request
+   * @returns {Promise.<UserModel[]>} - an array of updated users
+   */
+  public async updateUserAccessLevels(accessUpdates, emails: string[],
+    currentUser: UserModel): Promise<PrivateProfile[]> {
+    return this.transactions.readWrite(async (txn) => {
+      // Check for duplicate emails in the request
+      const emailSet = emails.reduce((set, email) => {
+        if (set.has(email)) {
+          return set;
+        }
+        set.add(email);
+        return set;
+      }, new Set<string>());
+
+      if (emailSet.size !== emails.length) {
+        throw new BadRequestError('Duplicate emails found in request');
+      }
+
+      // Strip out the user emails & validate that users exist
+      const userRepository = Repositories.user(txn);
+      const users = await userRepository.findByEmails(emails);
+      const emailsFound = users.map((user) => user.email);
+      const emailsNotFound = emails.filter((email) => !emailsFound.includes(email));
+
+      if (emailsNotFound.length > 0) {
+        throw new BadRequestError(`Couldn't find accounts matching these emails: ${emailsNotFound}`);
+      }
+
+      const updatedUsers = await Promise.all(accessUpdates.map(async (accessUpdate, index) => {
+        const { user, newAccess } = accessUpdate;
+
+        const matchingAccess = Object.keys(UserAccessType).find((access) => access === newAccess);
+        const updatingAccess = UserAccessType[matchingAccess];
+
+        const currUser = await userRepository.findByEmail(user);
+
```

Comment:
This `findByEmail` call is not needed, since on line 222 we already fetch all the users by their emails. Let's create a map of [email, UserModel] from the `findByEmails` call above so that we don't need to read from the database here